### DISCUSSION TITLE
[V4] Access to antl instance in mail view rendering

### DISCUSSION THREAD

**Discussion by maxgalbu** [adonisjs/core#2261] [Help] [answered]
I'd like to internationalize emails send to users, is there some way to access @adonisjs/antl in the mail view context? Do i need to pass the instance myself like this (not tested), or is there a better way?

```
await Mail.send('emails-welcome', {antl}, (message) => {
            message
                .to(email)
                .from('test@email.com')
                .subject('Welcome')
        });
```

**Comment by thetutlage** [comment, marked as answer]
@maxgalbu Any reason you think that passing `antl` manually isn't a great approach. Coz I will do the same. 

During an HTTP request, we create the `antl` instance by finding the user browser language. Whereas, the emails are usually sent in background and AdonisJS has no way to detect the browser language. You have to manually create the `antl` instance for the user language yourself and pass it to the template files

**Reply by maxgalbu** [reply]
@thetutlage thanks, this makes sense. 

In any case - correct me if I'm wrong - I could continue using that already created Antl instance and just switchLocale() on it, then pass it down to the templates. This is because I don't need to use it again, since I'm just returning a JSON. Correct?

**Reply by thetutlage** [reply]
You must use the `forLocale` method instead. It returns an ally instance for that locale. https://adonisjs.com/docs/4.1/internationalization#_switch_locale

```ts
const antlForFrench = Antl.forLocale('fr')

View.render('some-view', { antl: antlForFrench })
```
